client_test.go

```diff
@@ -22,27 +42,26 @@ func testTOPS(t *testing.T, symbols []string) {
 	if len(result) != len(symbols) {
 		t.Fatalf("Received %v results, expected %v", len(result), len(symbols))
 	}
-
-	// TODO(palpant): Test parsing with a mock http client.
-}
-
-func TestTOPS_OneSymbol(t *testing.T) {
-	testTOPS(t, []string{"SPY"})
 }
 
-func TestTOPS_TwoSymbols(t *testing.T) {
-	testTOPS(t, []string{"SPY", "AAPL"})
+var testCases = []struct {
+	symbols []string
+	code    int
+	body    string
+	err     error
+	headers map[string]string
+}{
```

Comment:
Since it is mocked, we can add an `expected` field and then in the test verify that it is parsed to the expected result (with [`reflect.DeepEqual`](https://golang.org/pkg/reflect/#DeepEqual)). Feel free to leave it as a TODO.

Reply:
Going to leave it as a TODO for now and come back to it! `Client.GetTOPS` returns `[]*TOPS` - do you mean having `expected` of the test struct as a list of TOPS with the content we expect to be parsed?

Also, would it also be worth doing to come back to the other test cases and redo them in a similar way? I'm not sure if this is the smoothest way to mock out a 3rd-party API, by copying a sample response as a string and using it in a mock, but if it's worth repeating I'll loop over the others too in another PR.

Reply:
> do you mean having expected of the test struct as a list of TOPS with the content we expect to be parsed?

Yep! It is somewhat pedantic for simple cases, but could help catch bugs like this one [ac6356b5](https://github.com/timpalpant/go-iex/commit/ac6356b566b0af1c3c626a6ac40322d432f5c485#diff-e2cd38f38509c9033189fbf0317e414e) where the response was not decoded correctly because the struct members were misnamed/mistyped. Since the data is mocked and fixed we can have stronger assertions.

> Also, would it also be worth doing to come back to the other test cases and redo them in a similar way?

Yeah, this way seems better to me. Having the mocked data in the tests makes it more self-contained, and better for CI. The assumption we rely on is that the API responses will not change, but that seems like a reasonable expectation. The `httptest` package seems like it was built for this use case.